tests/cli/test_cli.py

```diff
@@ -805,23 +806,34 @@ def test_upload1(ctx):
         _run_command([cl, 'uedit', 'codalab', '--disk-quota', ctx.disk_quota])
 
         # Run the same tests when on a non root user
-        user_name = 'non_root_user_' + random_name()
-        create_user(ctx, user_name, disk_quota='2000')
-        switch_user(user_name)
-        # expect to fail when we upload something more than 2k bytes
-        check_contains(
-            "Attempted to upload bundle of size 10.0k with only 2.0k remaining in user\'s disk quota",
-            _run_command(
-                [cl, 'upload', test_path('codalab.png')] + suffix,
-                expected_exit_code=1,
-                # To return stderr, we need to include
-                # the following two arguments:
-                include_stderr=True,
-                force_subprocess=True,
-            ),
-        )
-        # Switch back to root user
-        switch_user('codalab')
+        if not os.getenv('CODALAB_PROTECTED_MODE'):
+            # This test does not work when protected_mode is True.
+            user_name = 'non_root_user_' + random_name()
+            create_user(ctx, user_name, disk_quota='10')
+            # group_uuid = _run_command([cl, 'gnew', user_name])
+            switch_user(user_name)
+            worksheet_uuid = _run_command([cl, 'work', '-u'])
+            switch_user('codalab')
+            _run_command([cl, 'wperm', worksheet_uuid, 'public', 'a'])
+            switch_user(user_name)
+            _run_command([cl, 'work', worksheet_uuid])
+            # expect to fail when we upload something more than 2k bytes
+            check_contains(
+                'Upload aborted. User disk quota exceeded. '
```

Comment:
Just check if it contains "User disk quota exceeded" -- this way we don't need to change this test if we slightly modify the error message in the future.

Reply:
Done.